### ISSUE TITLE
Parse error: syntax error, unexpected '\'

### ISSUE THREAD

**Issue by rikless** [mockery/mockery#243]
I was using Mockery "mockery/mockery": "dev-master@dev", and after last composer update, all tests break with this message : 

```
PHP Parse error:  syntax error, unexpected '\' (T_NS_SEPARATOR), expecting identifier (T_STRING) in /path/vendor/mockery/mockery/library/Mockery/Loader/EvalLoader.php(15) : eval()'d code on line 25

Parse error: syntax error, unexpected '\' (T_NS_SEPARATOR), expecting identifier (T_STRING) in /path/vendor/mockery/mockery/library/Mockery/Loader/EvalLoader.php(15) : eval()'d code on line 25
PHP Fatal error:  Uncaught exception 'ReflectionException' with message 'Class Mockery_0__Eloquent does not exist' in /path/vendor/mockery/mockery/library/Mockery/Container.php:401
Stack trace:
#0 /path/vendor/mockery/mockery/library/Mockery/Container.php(401): ReflectionClass->__construct('Mockery_0__Eloq...')
#1 /path/vendor/mockery/mockery/library/Mockery/Container.php(193): Mockery\Container->_getInstance('Mockery_0__Eloq...', NULL)
#2 [internal function]: Mockery\Container->mock('Eloquent', 'Category')
#3 /path/vendor/mockery/mockery/library/Mockery.php(70): call_user_func_array(Array, Array)
#4 /path/app/tests/controllers/CategoriesTest.php(11): Mockery::mock('Eloquent', 'Category')
#5 /usr/share/pear/PHPUnit/Framework/TestSuite.php(473): CategoriesTest->__construct()
#6 /usr/share/pear/PHPUnit/Framework/TestS in /path/vendor/mockery/mockery/library/Mockery/Container.php on line 401
```

The Category class is in a namespace, loading from psr4, and extend the Eloquent class (from Laravel).

This was the test :

https://gist.github.com/anonymous/9531edc5c447f0cbac08#file-mackery

Moving to stable release fix the issue.

**Comment by UlrichEckhardt** [comment]
FWIW, I think you can 

    use MyNameSpace\Category;

and then refer to the class via 

    this->mock = m::mock('Eloquent', Category::class);

PHP does the proper magic there and turns the expression into the fully qualified class name. This works since PHP 5.5 or 5.6, I think.

**Comment by robertbasic** [comment]
@rikless I know it's been a couple of years (sorry about that!), but is this still a problem for you? If yes, can you provide a repository or some such where this issue is apparent? If no, then maybe we can close this issue, as I don't think anyone else reported similar issues since.

Thanks!

**Comment by rikless** [comment]
@robertbasic Can't really tell you, I don't use it right now, and since this post. Really sorry :/

**Comment by robertbasic** [comment]
The minimum test case for this is:

``` php
$m = \Mockery::mock('Baz', 'Spam');
```

and it got broken with ab2c696982fa47bb21d3bff1cd97b6b657d6f4f4 and fixed with 43420634f9504a05fb1ee03eedfb3d632ccbaf9e in #629 

git bisect ftw!
